includes/express-checkout/class-wc-payments-express-checkout-ajax-handler.php

```diff
@@ -55,303 +49,17 @@ public function init() {
-				}
-			}
-
-			// Force quantity to 1 if sold individually and check for existing item in cart.
-			if ( $product->is_sold_individually() ) {
-				$qty = apply_filters( 'wcpay_payment_request_add_to_cart_sold_individually_quantity', 1, $qty, $product_id, $variation_id );
-			}
-
-			if ( ! $product->has_enough_stock( $qty ) ) {
-				/* translators: 1: product name 2: quantity in stock */
-				throw new Exception( sprintf( __( 'You cannot add that amount of "%1$s"; to the cart because there is not enough stock (%2$s remaining).', 'woocommerce-payments' ), $product->get_name(), wc_format_stock_quantity_for_display( $product->get_stock_quantity(), $product ) ) );
-			}
-
-			$price = $this->express_checkout_button_helper->get_product_price( $product, $is_deposit, $deposit_plan_id );
-			$total = $qty * $price + $addon_value;
-
-			$quantity_label = 1 < $qty ? ' (x' . $qty . ')' : '';
-
-			$data  = [];
-			$items = [];
-
-			$items[] = [
-				'label'  => $product->get_name() . $quantity_label,
-				'amount' => WC_Payments_Utils::prepare_amount( $total, $currency ),
-			];
-
-			$total_tax = 0;
-			foreach ( $this->express_checkout_button_helper->get_taxes_like_cart( $product, $price ) as $tax ) {
-				$total_tax += $tax;
-
-				$items[] = [
-					'label'   => __( 'Tax', 'woocommerce-payments' ),
-					'amount'  => WC_Payments_Utils::prepare_amount( $tax, $currency ),
-					'pending' => 0 === $tax,
-				];
-			}
-
-			if ( wc_shipping_enabled() && $product->needs_shipping() ) {
-				$items[] = [
-					'label'   => __( 'Shipping', 'woocommerce-payments' ),
-					'amount'  => 0,
-					'pending' => true,
-				];
-
-				$data['shippingOptions'] = [
-					'id'     => 'pending',
-					'label'  => __( 'Pending', 'woocommerce-payments' ),
-					'detail' => '',
-					'amount' => 0,
-				];
-			}
-
-			$data['displayItems'] = $items;
-			$data['total']        = [
-				'label'   => $this->express_checkout_button_helper->get_total_label(),
-				'amount'  => WC_Payments_Utils::prepare_amount( $total + $total_tax, $currency ),
-				'pending' => true,
-			];
-
-			$data['needs_shipping'] = wc_shipping_enabled() && 0 !== wc_get_shipping_method_count( true ) && $product->needs_shipping();
-			$data['currency']       = strtolower( get_woocommerce_currency() );
-			$data['country_code']   = substr( get_option( 'woocommerce_default_country' ), 0, 2 );
-			$data['has_free_trial'] = class_exists( 'WC_Subscriptions_Product' ) ? WC_Subscriptions_Product::get_trial_length( $product ) > 0 : false;
-
-			wp_send_json( $data );
-		} catch ( Exception $e ) {
-			if ( is_a( $e, Invalid_Price_Exception::class ) ) {
-				Logger::log( $e->getMessage() );
-			}
-			wp_send_json( [ 'error' => wp_strip_all_tags( $e->getMessage() ) ], 500 );
-		}
+		add_filter( 'rest_pre_dispatch', [ $this, 'tokenized_cart_store_api_address_normalization' ], 10, 3 );
+		add_filter( 'woocommerce_get_country_locale', [ $this, 'modify_country_locale_for_express_checkout' ], 20 );
```

Comment:
Nitpick: With this clean-up, these are two remaining filters, then the name `AJAX_Handler` seems no longer relevant.

Reply:
Only the `ajax_add_to_cart` method in this class is really to handle AJAX calls. 
All other methods are focusing on the Store API request. So something more specific like `*_Rest_Store_Api_Handler` would be better.

Reply:
Also, `ajax_add_to_cart` here is now only to be used with WooPay: 

https://github.com/Automattic/woocommerce-payments/blob/88f7abbce85229e3978c60773a12c0f75f20e677/includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php#L87-L89

I think we can move this `ajax_add_to_cart` to `WC_Payments_WooPay_Button_Handler` and called it in `WC_Payments_WooPay_Button_Handler::init()`.